src/libsync/vfs/cfapi/vfs_cfapi.cpp

```diff
@@ -266,17 +266,34 @@ Vfs::AvailabilityResult VfsCfApi::availability(const QString &folderPath)
     return AvailabilityError::NoSuchItem;
 }
 
-void VfsCfApi::cancelHydration(const QString &requestId, const QString & /*path*/)
+Optional<HydrationJob *> VfsCfApi::findHydrationJob(const QString &requestId)
 {
-    // Find matching hydration job for request id
     const auto hydrationJobsIter = std::find_if(d->hydrationJobs.cbegin(), d->hydrationJobs.cend(), [&](const HydrationJob *job) {
         return job->requestId() == requestId;
     });
 
-    // If found, cancel it
     if (hydrationJobsIter != d->hydrationJobs.cend()) {
-        (*hydrationJobsIter)->cancel();
+        return *hydrationJobsIter;
     }
+
+    return {};
+}
+
+void VfsCfApi::cancelHydration(const QString &requestId, const QString & /*path*/)
+{
+    const auto hydrationJob = findHydrationJob(requestId);
+    if (hydrationJob) {
+        (*hydrationJob)->cancel();
+    }
+}
+
+bool VfsCfApi::hydrationCanceled(const QString &requestId)
```

Comment:
@FlexW Maybe better pick another name? Something makes me think of this method as a signal. `isHydrationCanceled` may wor too? Or maybe, `checkHydrationCanceled`?

Reply:
Yes, this name is bad. `I go with checkHydrationCanceled()`